listenbrainz_spark/recommendations/tests/test_candidate.py

```diff
@@ -259,3 +259,193 @@ def test_save_candidate_sets(self):
+                similar_artist_name='john',
+                user_name='vansika'
+            ),
+            schema=None
+        ))
+
+        df = utils.create_dataframe(
+            Row(
+                top_artist_credit_id=2,
+                top_artist_name="blahblah",
+                mb_artist_credit_id=1,
+                mb_artist_credit_mbids=['xxx'],
+                mb_recording_mbid='yyy',
+                msb_artist_credit_name_matchable='blahblah',
+                msb_recording_name_matchable='looloo',
+                recording_id=2,
+                user_name='vansika_1'
+            ),
+            schema=None
+        )
+
+        df = df.union(utils.create_dataframe(
+            Row(
+                top_artist_credit_id=2,
+                top_artist_name="Less Than Jake",
+                mb_artist_credit_id=1,
+                mb_artist_credit_mbids=['xxx'],
+                mb_recording_mbid='yyy',
+                msb_artist_credit_name_matchable='lessthanjake',
+                msb_recording_name_matchable='lalal',
+                recording_id=2,
+                user_name='vansika'
+            ),
+            schema=None
+        ))
+
+        top_artist_candidate_set_df_html = df.union(utils.create_dataframe(
+            Row(
+                top_artist_credit_id=1,
+                top_artist_name="Less Than Jake",
+                mb_artist_credit_id=1,
+                mb_artist_credit_mbids=['xxx'],
+                mb_recording_mbid='yyy',
+                msb_artist_credit_name_matchable='lessthanjake',
+                msb_recording_name_matchable='lalal',
+                recording_id=2,
+                user_name='vansika',
+            ),
+            schema=None
+        ))
+
+        df = utils.create_dataframe(
+            Row(
+                similar_artist_credit_id=2,
+                similar_artist_name="blahblah",
+                mb_artist_credit_id=1,
+                mb_artist_credit_mbids=['xxx'],
+                mb_recording_mbid='yyy',
+                msb_artist_credit_name_matchable='blahblah',
+                msb_recording_name_matchable='looloo',
+                recording_id=2,
+                user_name='vansika_1'
+            ),
+            schema=None
+        )
+
+        similar_artist_candidate_set_df_html = df.union(utils.create_dataframe(
+            Row(
+                similar_artist_credit_id=1,
+                similar_artist_name="Less Than Jake",
+                mb_artist_credit_id=1,
+                mb_artist_credit_mbids=['xxx'],
+                mb_recording_mbid='yyy',
+                msb_artist_credit_name_matchable='lessthanjake',
+                msb_recording_name_matchable='lalal',
+                recording_id=2,
+                user_name='vansika',
+            ),
+            schema=None
+        ))
+
+        recieved_user_data = candidate_sets.get_candidate_html_data(similar_artist_candidate_set_df_html,
```

Comment:
received

Reply:
Ughhhh.

Reply:
I swear I know the spelling. You believe me right?

Reply:
Doesn't really matter does it? 😆